runtime/src/iree/base/internal/math.h

```diff
@@ -307,4 +307,10 @@ static inline uint16_t iree_math_f32_to_f16(const float f32_value) {
   return (uint16_t)(sign | exp | mantissa);
 }
 
+// Rounds of 32-bit C `float` value to nearest 16-bit value and returns
+// 32-bit `float`
+static inline float iree_math_round_to_nearest_f16(const float f32_value) {
```

Comment:
add a TODO about the possibility of someday writing a faster impl of this function (it's called literally as many times as there are scalar (add or mul) ops in the reference matmuls, so it'll likely be quite significant on profiles, and e2e matmul tests latency are never far below the threshold where we feel compelled to do something about them, in fact we have repeatedly trimmed them in the past to keep their latency in check).

Reply:
There is a fast implementation of [half_t data type](https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/half.h) which uses fast conversion [(F32->F16) implementation](https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/half.h#L204) Support: 

- [CPUs with F16c enabled](https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/half.h#L216)
- [NVIDIA GPUs](https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/half.h#L213)
- [Software implementation as a fall back](https://github.com/NVIDIA/cutlass/blob/master/include/cutlass/half.h#L223)

I champion using this implementation of half data type until it becomes available in c++ standard.